src/componententity.cpp

```diff
@@ -351,8 +343,10 @@ UnitsPtr ComponentEntity::takeUnits(const std::string &name)
 
 void ComponentEntity::replaceUnits(size_t index, const UnitsPtr &units)
 {
-    removeUnits(index);
-    mPimpl->mUnits.insert(mPimpl->mUnits.begin() + index, units);
+    if (index < mPimpl->mUnits.size()) {
+        removeUnits(index);
+        mPimpl->mUnits.insert(mPimpl->mUnits.begin() + index, units);
+    }
```

Comment:
I would instead expect this function now to look something like (if we are to respect what we agreed in #140):

```
bool ComponentEntity::replaceUnits(size_t index, const UnitsPtr &units)
{
    if (removeUnits(index)) {
        mPimpl->mUnits.insert(mPimpl->mUnits.begin() + index, units);

        return true;
    } else {
        return false;
    }
}
```

The same holds for all the functions where we don't (indirectly) throw an exception anymore.